### PR TITLE
LPS-195531 - Allow admins to select filters defined as client extensions

### PR DESCRIPTION
**Pull request by bryceosterhaus**
https://liferay.atlassian.net/browse/LPS-182558

Depends on and rebased on top of  https://github.com/liferay-frontend/liferay-portal/pull/3658

**Preview:**

https://github.com/liferay-frontend/liferay-portal/assets/6843530/3064b1ac-876c-42d8-888e-516ac556cf50

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:frontend-data-set

**Comment by mateusralv** [comment]
Hi @bryceosterhaus I disregarded the failures related to the Data Set filters because the tests are ignored in the master and I also found some related.

#### Not possible to add a DataSet Fragment on the page

https://github.com/liferay-frontend/liferay-portal/assets/44584050/e1e40083-9dcc-4239-984a-dbdde5cc8b2f

[Testray Comparison 
](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1694809246746&testrayRunIdA=2424231664&testrayRunIdB=2424440076)

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:frontend-data-set

**Comment by mateusralv** [comment]
The failures in ci:test:frontend-data-set are not related 

[Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1695170421139&testrayRunIdA=2427188346&testrayRunIdB=2427353041)

File: modules/apps/object/object-api/src/main/java/com/liferay/object/definition/util/ObjectDefinitionUtil.java [hunk 1]
```diff
@@ -103,6 +103,9 @@ private static String _getInvokerFileNameRegex(
 			"Bookmark", "/bookmarks"
 		).put(
 			"FDSAction", "/data-set-manager/actions"
+		).put(
+			"FDSClientExtensionFilter",
+			"/data-set-manager/client-extension-filters"
```
Comment (dsanz) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/definition/util/ObjectDefinitionUtil.java:108]: Good catch!

File: modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-web/src/main/resources/META-INF/resources/admin/report/input_repeat.jsp [hunk 2]
```diff
@@ -182,31 +182,27 @@ int yearlyType = ParamUtil.getInteger(request, "yearlyType");
 <aui:script require="frontend-js-web/index as frontendJsWeb">
 	var {delegate} = frontendJsWeb;
 
-	var tables = document.querySelectorAll('#<portlet:namespace />recurrenceTypeDailyTable, #<portlet:namespace />recurrenceTypeMonthlyTable, #<portlet:namespace />recurrenceTypeNeverTable, #<portlet:namespace />recurrenceTypeWeeklyTable, #<portlet:namespace />recurrenceTypeYearlyTable');
+	var tables = document.querySelectorAll(
```
Comment (dsanz) [inline modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-web/src/main/resources/META-INF/resources/admin/report/input_repeat.jsp:185]: why this here? it's just SF right?

Comment (bryceosterhaus) [reply modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-web/src/main/resources/META-INF/resources/admin/report/input_repeat.jsp:185]: Yep, just SF

**Review by markocikos — COMMENTED** [review]
@bryceosterhaus Mostly looks good, some minor stuff in comments.

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -11417,6 +11418,7 @@ no-channel=No Channel
 no-channels-were-found=No channels were found.
 no-city-and-state-pairs-were-found-associated-with-the-given-zip=No city and state pairs were found associated with the given zip.
 no-classification-rules-are-configured-for-this-device-family=No classification rules are configured for this device family.
+no-client-extentions-are-available-create-a-frontend-data-set-filter-in-order-to-use-it=No Client Extensions are available. Create a Frontend Data Set filter in otder to use it.
```
[the remark this answers is not in the document]
Comment (bryceosterhaus) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:11421]: Good catch!

not sure about the first sentence thing... bchan had me correct one recently that wasn't long enough, so now I just make it more verbose and he hasn't denied those

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/141054

**Pull request closed without merging**